cst438_register/src/main/java/com/cst438/controller/StudentController.java

```diff
@@ -0,0 +1,84 @@
+package com.cst438.controller;
+
+import com.cst438.domain.Student;
+import com.cst438.domain.StudentDTO;
+import com.cst438.domain.StudentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class StudentController {
+
+    @Autowired
+    StudentRepository studentRepository;
+
+    @PostMapping("/student")
+    @Transactional
+    @CrossOrigin(origins = {"http://localhost:3000", "https://registerf-cst438.herokuapp.com/"})
+    
+    public Student addStudent (@RequestBody StudentDTO studentDTO) {
+        
+        Student existStudent = studentRepository.findByEmail(studentDTO.studentEmail);
+
+        if (existStudent != null) {
+            throw  new ResponseStatusException(HttpStatus.BAD_REQUEST, "Student email already exists.");
+        } 
+        else if (studentDTO.studentName == null || studentDTO.studentEmail == null)

```

Comment:
Code ensures that the body along with the HTTP post request has the sufficient amount of information, while also letting the user know what they are missing in their inputs